lenskit-core/src/main/java/org/lenskit/data/entities/AbstractEntity.java

```diff
@@ -57,6 +57,16 @@ public EntityType getType() {
     /**
      * {@inheritDoc}
      *
+     * This implementation delegates to {@link #hasAttribute(String)}
+     */
+    @Override
+    public boolean hasAttribute(TypedName<?> name) {
+        return hasAttribute(name.getName());
```

Comment:
I like the symmetry between `has=false` and `get=NoSuchAttrException` but right now we do not have the symetry that`has=true` then `get` will not fail, which I find more important. As you say, these exceptions should never happen in code and are basically unrecoverable (excepting shenanigans).